--- a/src/lib.rs
+++ b/src/lib.rs
@@ -1,7 +1,6 @@
 //! Implementation details of the `include_dir`.
 //!
 //! You probably don't want to use this crate directly.
-#![cfg_attr(feature = "nightly", feature(track_path, proc_macro_tracked_env))]
 
 use proc_macro::{TokenStream, TokenTree};
 use proc_macro2::Literal;
@@ -248,19 +247,11 @@
     s.split_at(index)
 }
 
-#[cfg(feature = "nightly")]
-fn get_env(variable: &str) -> Option<String> {
-    proc_macro::tracked_env::var(variable).ok()
-}
-
-#[cfg(not(feature = "nightly"))]
 fn get_env(variable: &str) -> Option<String> {
     std::env::var(variable).ok()
 }
 
 fn track_path(_path: &Path) {
-    #[cfg(feature = "nightly")]
-    proc_macro::tracked_path::path(_path.to_string_lossy());
 }
 
 #[cfg(test)]
